### PR TITLE
Chore/initial php stan fixes

### PR DESCRIPTION
**Pull request by bensquire**
@adamjimenez This PR is a hodge podge of changes, but I'm working towards the greater goal.

- Added return, param types where possible
- Coding standards
- Started work on adding an interface to components
- Started work on updating components to use modern file naming convention

I just don't want to get carried away and end up with the mother of all PRs. My next step is to update the `field` of each compenet to something like `render` that returns a string. This will help with thing such as unit testing.

### CODE REVIEW

**Comment by bensquire** [comment]
In good news I'm getting used to the CMS structure...

File: _lib/cms/components/Checkboxes.php [hunk 1]
```diff
@@ -182,12 +184,11 @@ public function format_value($value, $field_name)
     }
 
     // generates sql code for use in where statement
-    public function conditions_to_sql($field_name, $value, $func = '', $table_prefix = '')
+    public function conditions_to_sql($field_name, $value, $func = '', $table_prefix = ''): string
     {
-        return null;
```
Comment (adamjimenez) [inline _lib/cms/components/Checkboxes.php:187]: checkboxes returns null deliberately because the search uses a table join instead.

Comment (bensquire) [reply _lib/cms/components/Checkboxes.php:187]: Could we return string or null? `:?string`

Comment (adamjimenez) [reply _lib/cms/components/Checkboxes.php:187]: this component should always return null as the search is handled here:
https://github.com/adamjimenez/shiftlib/blob/master/_lib/cms/cms.php#L309

This should be re-factored but we'll need to think about how we handle the joins..

Comment (bensquire) [reply _lib/cms/components/Checkboxes.php:187]: This should be fixed now

**Comment by adamjimenez** [comment]
Renaming the components will break with existing configs unless there is some translation in place:
https://github.com/adamjimenez/shiftlib/blob/master/_lib/cms/_tpl/configure.php#L1013
and
https://github.com/adamjimenez/shiftlib/blob/master/_lib/cms/cms.php#L723

It will also break anywhere the field type is hardcoded e.g. throughout cms.php and other places like
https://github.com/adamjimenez/shiftlib/blob/master/_lib/cms/_tpl/default_edit.php#L131

**Comment by bensquire** [comment]
@adamjimenez This should be working now. Can we review etc and merge before making anymore changes?

**Pull request merged**